### PR TITLE
Migrate GauGAN to Keras3[TF Backend Only]

### PR DESCRIPTION
**Pull request by sineeli**
Migrate GuaGAN to keras3 as per [Keras CV Fix it](https://github.com/keras-team/keras-cv/issues/2211)

### CODE REVIEW

File: examples/generative/gaugan.py [hunk 1]
```diff
@@ -65,14 +65,12 @@
...
-for training our GauGAN model. Let's first download it. We also install
-TensorFlow Addons.
+for training our GauGAN model. Let's first download it.
 """
 
 """shell
-gdown https://drive.google.com/uc?id=1q4FEjQg1YSb4mPx2VdxL7LXKYu3voTMj
-unzip -q facades_data.zip
-pip install -qqq tensorflow_addons
+!wget https://drive.google.com/uc?id=1q4FEjQg1YSb4mPx2VdxL7LXKYu3voTMj -O facades_data.zip
```
Comment (fchollet) [inline examples/generative/gaugan.py:72]: Remove the `!`, it gets added automatically

File: examples/generative/gaugan.py [hunk 2]
```diff
@@ -623,8 +634,8 @@ def train_discriminator(self, latent_vector, segmentation_map, real_image, label
         with tf.GradientTape() as gradient_tape:
```
Comment (fchollet) [inline examples/generative/gaugan.py:634]: The code only works with the TF backend, right? In that case we should add a leading code snippet to set the backend to TF via `KERAS_BACKEND`.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/generative/gaugan.py (hunk 2 above)
Comment (sineeli) [reply examples/generative/gaugan.py:634]: Yes, it works only with tensorflow backend, by default Keras 3 directs to Tensorflow that's why didn't included the `KERAS_BACKEND`. If necessary I will include that now and regenerate. Thanks for the review!

Comment (fchollet) [reply examples/generative/gaugan.py:634]: Yes, please include it.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**